pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go

```diff
@@ -296,8 +298,12 @@ func createSelfSubjectAccessReviews(
 	ctx context.Context,
 	kubeClient kubernetes.Interface,
 	selfSubjectAccessReviews []authorizationv1.SelfSubjectAccessReview) (bool, *authorizationv1.SelfSubjectAccessReview, error) {
+	// delay a random seconds(between 1-10) for every time creation, because too many managed clusters registration on hub and send ssar requests at once could cause resource pressure
+	time.Sleep(time.Second * time.Duration(rand.Intn(10)))
```

Comment:
how about delay the each request? on the request loop

```
for i := range selfSubjectAccessReviews {
 
}
```

Reply:
or we may use the delay queue to handle this

https://github.com/kubernetes/client-go/blob/master/util/workqueue/delaying_queue.go

we add the each ssar request to a delay queue and handle them

Reply:
Assuming we are using delay queue like the following example, the concern is we can know the items in waitToAddQueue, we can only get items length of waitToProcessQueue.

```go
for i := range ssars {
  // ...
  dq.AddAfter(ssar[i], getRandomTime()) // dq is a delayqueue
}

for dq.Len() != 0 { // The Len() can only return the length of waitToProcessQueue, but we actually need the length of waitToAddQueue
   ssar := dq.Get()

  createRequest(ssar)

   dq.Done(item)
}
```

Reply:
Delay each request also works, but I'm not sure what a proper random time range should I set.
Right now, I set it to be a random number between 1-1000, and the random delay time should be 1ms - 1000ms.

Reply:
I think we need get the proper random time range with our test

if using delay queue, I think you two steps

1. add the ssar to the queue
2. start a go routine to handle the queue

Reply:
I would think you have a separated controller to handle sar only instead of in the status controller. It might be also reasonable to put it in the different condition from registration degraded.

Reply:
@qiujian16 create a new controller `klusterlet_hubconfig_controller` without delete `status_controller` yet. Please have a review of this new controller, if it is written as expected, then I can add testcases and remove duplicate part in status controller.

@skeeey Tried to start a go routine, but it still has the issue mentioned above --- Because items are added after a while, so we can't know for sure at a moment, whether all ssar are done or not.
Unless, we use waitgroup, but it will import complexity.